### ISSUE TITLE
Ingress conditions annotation doesn't create rule for http-headers

### ISSUE THREAD

**Issue by JayselDC** [kubernetes-sigs/aws-load-balancer-controller#1481]
I added the annotation to allow for custom rules on my listener but nothing gets created at all.

Here's the output of my ingress yaml
```apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/conditions.dummy: |
      {"Field":"http-header","HttpHeaderConfig":{"HttpHeaderName": "dummy", "Values":["test123"]}}
    alb.ingress.kubernetes.io/healthcheck-path: /
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/security-groups: sg-dummy
    alb.ingress.kubernetes.io/success-codes: 200,404
    alb.ingress.kubernetes.io/target-type: instance
    kubernetes.io/ingress.class: alb
  generation: 2
  name: dummy
  namespace: dummy
  resourceVersion: "75389781"
spec:
  rules:
  - host: port-1-dummy.net
    http:
      paths:
      - backend:
          serviceName: dummy
          servicePort: port-1
        path: /*
  - host: port-2-dummy.net
    http:
      paths:
      - backend:
          serviceName: dummy
          servicePort: port-2
        path: /*
  - host: port-3-dummy.net
    http:
      paths:
      - backend:
          serviceName: dummy
          servicePort: port-3
        path: /*
  - host: port-4-dummy.net
    http:
      paths:
      - backend:
          serviceName: dummy
          servicePort: port-4
        path: /*
status:
  loadBalancer:
    ingress:
    - hostname: dummy.elb.amazonaws.com
```

**Comment by M00nF1sh** [comment]
hi, what's your controller version?

**Comment by JayselDC** [comment]
> hi, what's your controller version?

v1.1.3

**Comment by phelanma** [comment]
I have also discovered this and lost a few days to this.

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: alb-ingress
  namespace: default
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/healthcheck-path: /health
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/inbound-cidrs: |-
     0.0.0.0/32 # sanitised data
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
    alb.ingress.kubernetes.io/auth-type: cognito
    alb.ingress.kubernetes.io/auth-scope: "email openid"
    alb.ingress.kubernetes.io/auth-session-timeout: '3600'
    alb.ingress.kubernetes.io/auth-session-cookie: AWSELBAuthSessionCookie
    alb.ingress.kubernetes.io/auth-on-unauthenticated-request: authenticate
    alb.ingress.kubernetes.io/load-balancer-attributes: idle_timeout.timeout_seconds=600
    alb.ingress.kubernetes.io/auth-idp-cognito: '{"UserPoolArn": "arn:aws:cognito-idp:aws_region:0000000000:userpool/aws_region_ABCSD","UserPoolClientId":"abcdefghijklmn00000","UserPoolDomain":"auth.mydomain.com"}'
    alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:aws_region:0000000000:certificate/AAA00112-3333-4444-abc4-000000ab0c00
    alb.ingress.kubernetes.io/actions.rule-webapp: >
      {"Type":"forward","ForwardConfig":{"TargetGroups":[{"ServiceName":"arene-ai-webapp","ServicePort":"80","Weight":100}],"TargetGroupStickinessConfig":{"Enabled":false,"DurationSeconds":0}}}
    alb.ingress.kubernetes.io/conditions.rule-xapi: >
      [{"Field":"http-header","HttpHeaderConfig":{"HttpHeaderName": "x-api-request", "Values":[true]}}]
    alb.ingress.kubernetes.io/actions.rule-xapi: >
      {"Type":"forward","ForwardConfig":{"TargetGroups":[{"ServiceName":"arene-ai-gateway","ServicePort":"8080","Weight":100}],"TargetGroupStickinessConfig":{"Enabled":false,"DurationSeconds":0}}}
spec:
  rules:
    - host: mydomain.com
      http:
        paths:
          - path: /*
            backend:
              serviceName: ssl-redirect
              servicePort: use-annotation
          - path: /*
            backend:
              serviceName: rule-xapi
              servicePort: use-annotation

          - path: /*
            backend:
              serviceName: rule-webapp
              servicePort: use-annotation

```

To add to the OP - in our case there are no errors or anything indicating it has failed or theres a problem. 
Just wait 2hrs an no ALB!

So we removed the conditions/actions and used:  
```
  - path: /*
            backend:
              serviceName: name-of-service1
              servicePort: 8080
- path: /*
            backend:
              serviceName: name-of-service2
              servicePort: 80
```
And it created the ALB, but as you can see this is not what we originally specified.

Kind of a blocking problem which is unfortunate because its been pretty damn useful!

**Comment by M00nF1sh** [comment]
@JayselDC 
Hi, sorry for late reply. The advanced routing feature is only available for 1.1.5.
If this issue still exists, please upgrade to our v2 version. 
Please reopen a new issue if the conditions annotation doesn't work in v2 version
